spyder/plugins/editor/widgets/codeeditor/tests/test_codeeditor.py

```diff
@@ -693,5 +694,35 @@ def test_cell_highlight(codeeditor, qtbot):
     assert editor.current_cell[0].selectionEnd() == 8
 
 
+@pytest.mark.skipif(
+    sys.platform.startswith("linux") and running_in_ci(),
+    reason="Fails on Linux and CI"
+)
+def test_shortcut_for_widget_is_updated(codeeditor, qtbot):
+    """Test shortcuts for codeeditor are updated on the fly."""
+    editor = codeeditor
+    text = ('aa\nbb\ncc\ndd\n')
+    editor.set_text(text)
+
+    # Check shortcuts were registered
+    assert editor._shortcuts != {}
+
+    # Check "move line down" shortcut is working as expected
+    qtbot.keyClick(editor, Qt.Key_Down, modifier=Qt.AltModifier)
+    assert editor.toPlainText() == "bb\naa\ncc\ndd\n"
+
+    # Change "move line down" to a different shortcut
+    editor.set_conf("editor/move line down", "Ctrl+B", section="shortcuts")
+    qtbot.wait(300)
+
+    # Check new shortcut works
+    qtbot.keyClick(editor, Qt.Key_B, modifier=Qt.ControlModifier)
+    assert editor.toPlainText() == "bb\ncc\naa\ndd\n"
+
+    # Check previous shortcut doesn't work
+    qtbot.keyClick(editor, Qt.Key_Down, modifier=Qt.AltModifier)
+    assert editor.toPlainText() == "bb\ncc\naa\ndd\n"
```

Comment:
Could be possible to add as part of this test or in a different test a full config restart and check that the original shortcut works?

Reply:
Good idea. I expanded this test to include resetting shortcuts.